DataFormats/Provenance/src/BranchDescription.cc

```diff
@@ -48,27 +48,24 @@ namespace edm {
   BranchDescription::BranchDescription(BranchType const& branchType,
                                        std::string const& moduleLabel,
                                        std::string const& processName,
-                                       std::string const& className,
-                                       std::string const& friendlyClassName,
                                        std::string const& productInstanceName,
-                                       TypeWithDict const& theTypeWithDict,
+                                       edm::TypeID const& theType,
                                        bool produced,
                                        bool availableOnlyAtEndTransition,
                                        std::set<std::string> const& aliases)
       : branchType_(branchType),
         moduleLabel_(moduleLabel),
         processName_(processName),
         branchID_(),
-        fullClassName_(className),
-        friendlyClassName_(friendlyClassName),
+        fullClassName_(theType.className()),
+        friendlyClassName_(theType.friendlyClassName()),
         productInstanceName_(productInstanceName),
         branchAliases_(aliases),
         transient_() {
     setDropped(false);
     setProduced(produced);
     setOnDemand(false);
     transient_.availableOnlyAtEndTransition_ = availableOnlyAtEndTransition;
-    setUnwrappedType(theTypeWithDict);
```

Comment:
If you are not calling this (I assume member function) should the function go away?

Reply:
It is still called elsewhere in the file, but given that it (and `setWrappedType()`) are **public** functions, presently used only in `BranchDescription.cc`, and the function(s) literally only set the value, I agree these functions should go away.